### ISSUE TITLE
`cast` method for applying changes from an attribute hash

### ISSUE THREAD

**Issue by faultyserver** [Crecto/crecto#208]
In my current application, I have a lot of methods that build up new instances of model objects and then do work with them. Currently (at least as far as I'm aware), the only way of doing this with Crecto directly is calling `Model.new` and then applying each parameter individually. I've seen some other libraries that provide auto constructors, though I'm not actually sure if they would work with `Changeset`s properly...

I'd like to see something similar to Ecto's [`cast/4`](https://hexdocs.pm/ecto/Ecto.Changeset.html#cast/4) for applying multiple changes to a model at once given an attribute hash. The main, immediate benefits that I see in this method are automatically applying multiple changes at once and "whitelisting" attributes that can be changed.

As a temporary solution, I've implemented this code as part of `Schema` (the only place I was able to access the `CRECTO_FIELDS` constant) and it seems to be working fairly well:

```crystal
def cast(attrs)
  attrs.each do |field, value|
    case field.to_s
      {% for model_field in CRECTO_FIELDS %}
      when "{{model_field[:name].id}}"
        self.{{model_field[:name].id}} = value.as({{model_field[:type].id}})
      {% end %}
    end
  end
  self.class.changeset(self)
end
```

This method returns a `Changeset` so that you can immediately check for errors easily. An example usage looks like this:

```crystal
class User < Crecto::Model
  schema "users" do
    field :name, String
    field :job_title, String
    field :age, Int32
  end

  validate_required :age
end

def create_user(**attrs)
  user = User.new.cast(attrs)
end

user = create_user(name: "Jon", job_title: "developer", irrelevant_field: 2351)

puts user.instance.to_json #=> {"name":"Jon","job_title":"developer"}
puts user.errors #=> [{:field => "age", :message => "is required"}]
```

Additionally, since `Repo.insert` and the like accept changesets as arguments, this naturally leads into a cast->insert pattern:

```crystal
user = create_user(name: "Jon", job_title: "developer", irrelevant_field: 2351)
if user.valid?
  Repo.insert(user)
end
```

**tl;dr** I'd like to port over Ecto's [`cast/4`](https://hexdocs.pm/ecto/Ecto.Changeset.html#cast/4) to make it easier to create and work with model instances. I've got a temporary solution in place for myself, but a full solution would need to better support things like associations and accepting changesets as arguments.

**Comment by fridgerator** [comment]
💯   I'm doing the same thing usually, adding each property individually.  I like this idea a lot.  There is `update_from_hash` https://github.com/Crecto/crecto/blob/master/src/crecto/schema.cr#L249 however it was put there for crecto-admin usage and doesn't work great ouside of that usecase.
